course02/homework01/gavin/list.h

```diff
@@ -0,0 +1,47 @@
+//
+// Created by gavin on 19/12/2018.
+//
+
+#include <cstdio>
+#include <memory>
+#include <iostream>
+struct Element
+{
+    Element(int data) : mData(data)
+    {}
+    int mData{};
+    std::shared_ptr<Element> mNext;
+    std::weak_ptr<Element> mPrevious;
```

Comment:
This works. Simpler and faster than that: unique_ptr for next, raw pointer for prev. I know I told you not to use raw pointers though, so I understand your choice. Why shared_ptr is not a great choice here: because the counter will always be one...